src/test/ui/did_you_mean/bad-assoc-ty.stderr

```diff
@@ -135,7 +135,7 @@ help: use type parameters instead
 LL | fn foo<X: K<T, T>, T>(x: X) {}
    |             ^  ^ ^^^
 
-error[E0121]: the type placeholder `_` is not allowed within types on item signatures
+error[E0121]: the type placeholder `_` is not allowed within functions
```

Comment:
Well, `_` types are certainly allowed within functions.

The old wording was more precise by talking about types in signatures.
I suggest to extend it with the item kind instead:
```rust
"the type placeholder `_` is not allowed within types on {kind} item signatures"
```

Reply:
I don't know how "not allowed within types on function(s) item signatures" is going to work, so I guess I should change it to "not allowed within types on item signatures for {kind}/functions"